ebos/eclmpiserializer.hh

```diff
@@ -147,6 +147,53 @@ public:
     }
 
 
+    //! \brief Handler for std::variant<> with two fundamental types
+
+    /*
+      This std::variant serialization is highly specialized:
+
+        1. It is hardcoded to take exactly two types T0 and T1.
+
+        2. Both T0 and T1 must be basic types where Mpi::pack(T, ...) overloads
+           must exist.
+
+    */
+    template<class T0, class T1>
+    void variant(const std::variant<T0,T1>& _data)
+    {
+        auto pack_size = [&](auto& d) {
+                             m_packSize += Mpi::packSize(d, m_comm);
+                         };
+
+        auto pack = [&](auto& d) {
+                          Mpi::pack(d, m_buffer, m_position, m_comm);
+                      };
+
+        std::variant<T0,T1>& data = const_cast<std::variant<T0,T1>&>(_data);
```

Comment:
The `variant` member function accepts temporaries (parameter type is reference to `const`) but this will is undefined behaviour if anyone passes a temporary to the `UNPACK` operation.  That problem is, of course, shared by all the other serialisation functions too,  but I'd really prefer if we didn't add more of it.